api/src/room_store.py

```diff
@@ -0,0 +1,23 @@
+import os
+import json
+
+
+class RoomStore:
+    def __init__(self):
+        self.path = os.environ['ROOM_STORE_DIR']
+        if not os.path.exists(self.path):
+            os.mkdir(self.path)
+
+    def get_all_room_ids(self) -> list:
+        return os.listdir(self.path)
+
+    def write_room_data(self, room: str, data: dict):
+        with open(f'{self.path}/{room}', 'w') as f:
```

Comment:
This is scary, if room ids ever become user-controllable (via a bug or otherwise) then room ids like `../../../` will allow malicious actors to overwrite important files. You should verify that the resulting path is a child of the room directory before writing.

See here for an example: https://security.openstack.org/guidelines/dg_using-file-paths.html

Reply:
Good catch, thanks